open_instruct/dpo_tune.py

```diff
@@ -1022,37 +1020,60 @@ def load_model():
                 optimizer.zero_grad()
                 lr_scheduler.step()
 
+                # We keep track of the loss at each logged step
+                with torch.no_grad():
+                    chosen_rewards = (args.dpo_beta * (policy_chosen_logps - reference_chosen_logps)).mean()
+                    rejected_rewards = (args.dpo_beta * (policy_rejected_logps - reference_rejected_logps)).mean()
+                    average_rewards = (chosen_rewards + rejected_rewards) / 2
+                    accuracy = (chosen_rewards > rejected_rewards).float().mean()
+                    margin = (chosen_rewards - rejected_rewards).mean()
+                    kl = average_rewards / args.dpo_beta
+                    local_metrics[0] += loss
+                    local_metrics[1] += chosen_rewards
+                    local_metrics[2] += rejected_rewards
+                    local_metrics[3] += average_rewards
+                    local_metrics[4] += accuracy
+                    local_metrics[5] += margin
+                    local_metrics[6] += policy_chosen_logps.mean()
+                    local_metrics[7] += policy_rejected_logps.mean()
+                    local_metrics[8] += kl
+                    if args.load_balancing_loss:
+                        local_metrics[19] += weighted_aux_loss
+
             # Checks if the accelerator has performed an optimization step behind the scenes
             if accelerator.sync_gradients:
                 progress_bar.update(1)
                 completed_steps += 1
-                if args.logging_steps and completed_steps % args.logging_steps == 0:
-                    avg_loss = (
-                        accelerator.gather(total_loss).mean().item()
-                        / args.gradient_accumulation_steps
-                        / args.logging_steps
-                    )
```

Comment:
I am not sure if the existing logic is correct. When `logging_steps = 1`, and when `completed_steps = 1`, we are dividing the `args.gradient_accumulation_steps` incorrectly because there has not been any accumulated gradient steps, right?